tokio/src/runtime/basic_scheduler.rs

```diff
@@ -214,12 +282,15 @@ where
     CURRENT.set(context, || f(scheduler, context))
 }
 
-impl<P> Drop for BasicScheduler<P>
-where
-    P: Park,
-{
+impl<P: Park> Drop for BasicScheduler<P> {
     fn drop(&mut self) {
-        enter(self, |scheduler, context| {
+        let mut inner = {
+            let mut lock = self.inner.lock().expect("BasicScheduler Inner lock");
+            lock.take()
+                .expect("Oh no! We never placed the Inner state back!")
```

Comment:
These `expect`s could double-panic if the thread is already panicking (and, I think, the `Mutex` _will_ be poisoned if we are panicking. Perhaps we should check if the thread is panicking and do...something else...if it is?